pandas/core/frame.py

```diff
@@ -555,6 +558,16 @@ def __init__(
                     values, index, columns, dtype=values.dtype, copy=False
                 )
 
+        if manager is None:
+            manager = get_option("mode.data_manager")
```

Comment:
Can you check if this slows down `DataFrame()` noticeably? I recall one of the options-related methods being somewhat slow.

Reply:
I ran a few timings (on this branch), but it's not fully clear:

```python
In [1]: from pandas._config import get_option  

In [2]: %timeit get_option("mode.data_manager")  
1.89 µs ± 173 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)

In [3]: %timeit pd.DataFrame(np.random.randn(3, 2))
86.9 µs ± 5.2 µs per loop (mean ± std. dev. of 7 runs, 10000 loops each)

In [4]: df = pd.DataFrame(np.random.randn(3, 2))   

# creating from manager (with manager type matching default of "array")
In [5]: %timeit pd.DataFrame(df._mgr) 
2.23 µs ± 465 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)

# specifing the manager -> don't look up the option, you would expect it to be faster (but not much difference)
In [7]: %timeit pd.DataFrame(df._mgr, manager="array") 
1.95 µs ± 75 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)

In [8]: df = pd.DataFrame(np.random.randn(3, 2), manager="block")   

In [9]: %timeit pd.DataFrame(df._mgr, manager="block") 
2.29 µs ± 360 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)

In [10]: pd.options.mode.data_manager = "block"  

In [11]: %timeit pd.DataFrame(df._mgr)  
2.33 µs ± 245 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)

In [12]: %timeit pd.DataFrame(df._mgr, manager="block")   
2.28 µs ± 246 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)

In [13]: %timeit get_option("mode.data_manager")  
1.79 µs ± 246 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)
```

So while the `get_option` itself takes a bit less than 2µs, avoiding it in the DataFrame constructor (by specifying the option), you don't really see an equivalent speed-up. 

In any case, I think it's only relevant for the case of passing a Manager, because once you pass any kind of data (like an array I did above), it takes much more time anyway.

Reply:
Ah, what I forgot in the explanation above, is that right now we don't check the option for the "fastpath" case of passing just a manager object. Only when data gets passed that needs to be converted to a manager (or when the manager needs to be reindexed), then the option is checked (and in this case the constructor takes a lot more time than checking the option anyway).